### DISCUSSION TITLE
Viewing previously received messages on device screen

### DISCUSSION THREAD

**Discussion by thatSFguy** [meshtastic/Discussions#156] [Firmware]
Hello Meshers!

Unfortunately [a pull request](https://github.com/meshtastic/firmware/pull/3784) which would have fulfilled some people's desire to view previous messages was auto closed by the bot and not merged.  

Long story short, the screen on most devices in underutilized and there is little value of actually having a screen besides to get your random BLE pin displayed when one pairs it with a phone.  I think it would be super helpful to be able to to just pick up your node and see what messages came in without checking the phone app or making sure BLE is connected.  

Some use cases for displaying previous messages on the device:
-  Reviewing messages when you come home
-  Loaning a node to someone without having to install the app to view messages 
-  Seeing messages on a mobile node (e.g. in your car without fumbling with your phone)
-  Building standalone meshtastic devices with a CardKB or similar.

I do have a T-Deck with the experimental UI, which i think is awesome, but if we could just get previous message piece brought over to the nodes most of us own it would open up more use cases, and allow us to bring technically challenged people along for the Meshtastic ride.

Here are a couple of screenshots from the closed pull request:
![meshtastic with previous message display3](https://github.com/user-attachments/assets/e2d0a6a1-bd1c-4a15-9bcb-9e7ff3bb7342)
![meshtastic with previous message display2](https://github.com/user-attachments/assets/b9e5bf98-e799-4f71-99c2-3ad98dbe7b41)
![meshtastic with previous message display](https://github.com/user-attachments/assets/94648f50-0602-4c6f-82b8-249beade0c02)

**Comment by garthvh** [comment]
That pull request was not going to be merged

**Reply by thatSFguy** [reply]
The merge request aside, do I repost this without the link to the PR for discussion purposes?

**Comment by thatSFguy** [comment]
I don't care if it was going to be merged or not.  The discussion point about having something that show the previous messages is still valid.
